### PR TITLE
Refactor/ixfor

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: src/Test/StateMachine/Internal/Sequential.hs [hunk 1]
```diff
@@ -70,16 +65,15 @@ liftGen gens pid ns returns ixfor = sized $ \sz -> runStateT (go sz) ns
       v <- choose (0, max 0 (u - 1))
       return . Just $ IntRef (Ref v) pid
 
-  go :: Int -> StateT (Map ix Int) Gen [Untyped' cmd ConstIntRef]
+  go :: Int -> StateT (Map ix Int) Gen [Untyped' cmd (ConstIntRef)]
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: Unnecessary `()`

File: src/Test/StateMachine/Utils.hs [hunk 2]
```diff
@@ -126,6 +126,19 @@ iall
 iall p = ifoldr (\i x ih -> p i x && ih) True
 
 class (IxFunctor t, IxFoldable t) => IxTraversable (t :: (TyFun ix * -> *) -> *) where
-  itraverse :: Applicative f => Proxy q
+
```
Comment (Danten) [inline src/Test/StateMachine/Utils.hs]: We should have a `{-# MINIMAL itraverse | ifor #-}`

File: example/src/DieHard.hs [hunk 3]
```diff
@@ -177,6 +176,15 @@ ixfor _ EmptySmall   _ = pure EmptySmall
 ixfor _ SmallIntoBig _ = pure SmallIntoBig
 ixfor _ BigIntoSmall _ = pure BigIntoSmall
 
+instance IxFunctor  (Untyped Step) where
+  ifmap _ _ = undefined
```
Comment (Danten) [inline example/src/DieHard.hs]: I'm contractually obliged to point out this `undefined`

File: example/src/DieHard.hs [hunk 4]
```diff
@@ -177,6 +176,15 @@ ixfor _ EmptySmall   _ = pure EmptySmall
...
+
+instance IxFoldable (Untyped Step) where
+  ifoldMap _ _ = undefined
```
Comment (Danten) [inline example/src/DieHard.hs]: And this one

File: src/Test/StateMachine/Internal/Sequential.hs (hunk 1 above)
Comment (stevana) [inline src/Test/StateMachine/Internal/Sequential.hs]: Thanks, fixed!

File: src/Test/StateMachine/Utils.hs (hunk 2 above)
Comment (stevana) [inline src/Test/StateMachine/Utils.hs]: Thanks, fixed!

File: src/Test/StateMachine/Utils.hs [hunk 5]
```diff
@@ -125,7 +125,21 @@ iall
   => (forall i. Sing (i :: ix) -> p @@ i -> Bool) -> t p -> Bool
 iall p = ifoldr (\i x ih -> p i x && ih) True
 
+{-# MINIMAL itraverse | ifor #-}
```
Comment (Danten) [inline src/Test/StateMachine/Utils.hs]: I think the pragma should be inside the class
https://downloads.haskell.org/~ghc/7.8.1/docs/html/users_guide/pragmas.html#minimal-pragma

File: example/src/DieHard.hs (hunk 3 above)
Comment (stevana) [inline example/src/DieHard.hs]: I know, I know... I'll try to fix this `IxFunctor *` and `IxFunctor1` stuff next.

File: src/Test/StateMachine/Utils.hs (hunk 5 above)
Comment (stevana) [inline src/Test/StateMachine/Utils.hs]: Ah! Thanks!

**Pull request merged**